### PR TITLE
WFLY-7028 Deprecate component=backup-for

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.jboss.org/browse/WFLY-7028

### CODE REVIEW

File: clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties [hunk 1]
```diff
@@ -357,6 +357,7 @@ infinispan.backup.bring-site-online=Re-enables a previously disabled backup site
 infinispan.backup.take-site-offline=Disables backup to a remote site.
 
 infinispan.component.backup-for=A cache for which this cache acts as a backup (for use with cross site replication).
+infinispan.component.backup-for.deprecated=Deprecated. Backup designation should match the current cache name.
```
Comment (rhusar) [inline clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties]: s/should/have to?

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/xsite/XSiteBackupForTestCase.java [hunk 2]
```diff
@@ -1,189 +0,0 @@
...
- *   LON:web:repl backed up by NYC:web:repl, SFO:web:LONrepl
- *   NYC not backed up
- *   SFO not backed up
- *
- *
- * @author Richard Achmatowicz
- */
-@RunWith(Arquillian.class)
-@RunAsClient
-public class XSiteBackupForTestCase extends ExtendedClusterAbstractTestCase {
```
Comment (rhusar) [inline testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/xsite/XSiteBackupForTestCase.java]: FYI I am already removing the test in https://github.com/wildfly/wildfly/pull/10697

**Review by rhusar — APPROVED** [review]

File: testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/xsite/XSiteBackupForTestCase.java (hunk 2 above)
Comment (pferraro) [reply testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/xsite/XSiteBackupForTestCase.java]: Then it's a race to see which gets merged first... :)

File: clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties (hunk 1 above)
Comment (pferraro) [reply clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties]: They are synonymous.

Comment (rhusar) [reply clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties]: Hm, I mean "must" then.

https://www.ietf.org/rfc/rfc2119.txt

Comment (pferraro) [reply clustering/infinispan/extension/src/main/resources/org/jboss/as/clustering/infinispan/subsystem/LocalDescriptions.properties]: OK - you win.

**Pull request merged**